cardano-testnet/src/Testnet/Util/Base.hs

```diff
@@ -7,16 +7,18 @@ module Testnet.Util.Base
 import           GHC.Stack (HasCallStack)
 import           System.Info (os)
 
+import qualified GHC.Stack as GHC
 import qualified Hedgehog as H
 import qualified Hedgehog.Extras.Test.Base as H
 
 
 integration :: HasCallStack => H.Integration () -> H.Property
-integration = H.withTests 1 . H.propertyOnce
+integration f = GHC.withFrozenCallStack $ H.withTests 1 $ H.propertyOnce f
 
-integrationRetryWorkspace :: Int -> FilePath -> (FilePath -> H.Integration ()) -> H.Property
-integrationRetryWorkspace n workspaceName f = integration $ H.retry n $ \i ->
-  H.runFinallies $ H.workspace (workspaceName <> "-" <> show i) f
+integrationRetryWorkspace :: HasCallStack => Int -> FilePath -> (FilePath -> H.Integration ()) -> H.Property
+integrationRetryWorkspace n workspaceName f = GHC.withFrozenCallStack $
+  integration $ H.retry n $ \i ->
+    H.runFinallies $ H.workspace (workspaceName <> "-" <> show i) f
```

Comment:
The `workspace` function reports the caller's module in the workspaces `module` file so that it is easy to find the failing test that created the workspace.

However, these functions are utility functions and we don't want to report `Testnet.Util.Base`.  We instead want to report the module of the functions that call these functions.  Hence we freeze the call stack.